### DISCUSSION TITLE
How to exclude defered statements?

### DISCUSSION THREAD

**Discussion by katexochen** [kisielk/errcheck#205] [Q&A] [answered]
Hi, how can I exclude defered statements? Thanks for helping!

```go
defer conn.Close() // this shouldn't throw an error
```

**Comment by dtcaciuc** [comment, marked as answer]
Regular way for excluding functions should work, see https://github.com/kisielk/errcheck#excluding-functions

I just successfully tried it with database/sql.DB

**Reply by katexochen** [reply]
Sorry, I think my question wasn't precise enough. I want to exclude **all** defered function calls, not a specific one.

**Reply by dtcaciuc** [reply]
Ok, I see. Errcheck doesn't currently support that, and to be honest, it doesn't sound like something we should add (at least to me). While some deferred error returns are safe to ignore, others aren't, and such feature will not allow to see the latter once enabled.

I obviously don't know the codebase you're working with, but I can't imagine the number of distinct deferred functions returning errors is so overwhelming that you can't explicitly list all of them. If those are methods implementing a common interface, then you can specify the interface type to ignore calls for everything that implements it. And, if those functions are part of your project, you can also simply change the function to not return an error and maybe log it instead?

Hope that helps.

**Reply by katexochen** [reply]
Thanks for your answer @dtcaciuc! I didn't noticed I could exclude interfaces, helped a lot. :+1:
